#!/usr/bin/env python
PACKAGE = "sub_trajectory"

from dynamic_reconfigure.parameter_generator_catkin import *
import yaml

gen = ParameterGenerator()

gen.add("thrust_limit", double_t, 0, "Maximum output of each thruster", 0.5, 0, 1)

data = []
with open("../../src/sub_trajectory/param/high_level_thrusters.param", "r") as file:
  data = yaml.load(file.read())

for k in sorted(data['thrusters']):
  gen.add(data['thrusters'][k]['thruster_name']+"_"+str(k), bool_t, 0, "Whether or not to use this thruster", True)

exit(gen.generate(PACKAGE, "vector_thrust", "Vector"))
